aries-site/src/layouts/navigation/Search.js

```diff
@@ -1,114 +1,180 @@
+    const { tags, title } = valueObj;
+    const nextOption = {
+      ...option,
+      value: { ...valueObj, matchType: [], matchLocation: [] },
+    };
+
+    /* Title Match */
+    if (regexp.test(title)) {
+      nextOption.value.matchType.push('ExactMatch');
+      nextOption.value.matchLocation.push('Title');
+    }
+    /* Content Match */
+    // number of matches should affect weighting
+    // TODO
+
+    /* Metadata: Tag Match */
+    else if (regexp.test(tags)) {
+      nextOption.value.matchType.push('ExactMatch');
+      nextOption.value.matchLocation.push('Tags');
+    }
+
+    return nextOption;
+  });
+
+  return results;
+};
+
+const search = (data, query) => {
+  const results = exactMatch(data, query).filter(
+    option => option.value.matchType.length > 0,
+  );
+
+  return sortSearch(results);
+};
+
+export const Search = ({ setOpen }) => {
   const router = useRouter();
   const size = useContext(ResponsiveContext);
   const [value, setValue] = useState('');
-  const [suggestions, setSuggestions] = useState(allSuggestions);
-  const inputRef = useRef();
+  const [suggestions, setSuggestions] = useState(
+    formatSearchResults(null, allSuggestions),
+  );
 
-  useEffect(() => {
-    if (focused && inputRef.current) {
-      inputRef.current.focus();
-    }
-  }, [focused, setFocused]);
+  const onClose = () => {
+    setOpen(false);
+  };
 
   const onChange = event => {
     const {
       target: { value: nextValue },
     } = event;
     let nextSuggestions;
-    if (nextValue) {
-      const regexp = new RegExp(nextValue, 'i');
-      nextSuggestions = allSuggestions.filter(c => regexp.test(c));
+    if (nextValue.length) {
+      nextSuggestions = search(allSuggestions, nextValue);
     } else {
       nextSuggestions = allSuggestions;
     }
-    // don't use newer value if nothing matches it
-    if (nextSuggestions.length > 0) {
-      setValue(nextValue);
-      setSuggestions(nextSuggestions);
-    }
+
+    setSuggestions(
+      formatSearchResults(
+        nextValue.length > 0 ? nextValue : null,
+        nextSuggestions,
+      ),
+    );
+    setValue(nextValue);
   };
 
   const onEnter = () => {
     if (value) {
       if (suggestions.length === 1) {
-        router.push(nameToPath(suggestions[0]));
+        router.push(nameToPath(suggestions[0].value.title));
       } else {
         const matches = allSuggestions.filter(
-          c => c.toLowerCase() === value.toLowerCase(),
+          c => c.label.toLowerCase() === value.toLowerCase(),
         );
         if (matches.length === 1) {
-          router.push(nameToPath(matches[0]));
+          router.push(nameToPath(matches[0].value.title));
         }
       }
     }
   };
 
   const onSelect = event => {
-    const href = nameToPath(event.suggestion);
+    const href = nameToPath(event.item.value.title);
     if (internalLink.test(href)) {
       router.push(href);
-      setFocused(false);
+      setOpen(false);
```

Comment:
```suggestion
      onClose();
```
would we want to do this here?

Reply:
done